ziti-springboot-client/src/main/java/org/openziti/springboot/client/web/config/ZitiHttpClientConfiguration.java

```diff
@@ -42,126 +40,125 @@
 import org.apache.hc.core5.pool.PoolReusePolicy;
 import org.apache.hc.core5.util.TimeValue;
 import org.openziti.Ziti;
+import org.openziti.ZitiContext;
 import org.openziti.springboot.client.web.httpclient.ZitiConnectionSocketFactory;
 import org.openziti.springboot.client.web.httpclient.ZitiSSLConnectionSocketFactory;
 import org.springframework.beans.factory.BeanCreationException;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.io.Resource;
 import org.springframework.http.client.ClientHttpRequestFactory;
 import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
 import org.springframework.web.client.RestTemplate;
+import jakarta.annotation.PreDestroy;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 @Configuration
 public class ZitiHttpClientConfiguration {
 
   // The default timeout when requesting a connection from the connection manager.
-  private static final long DEFAULT_CONNECTION_REQUEST_TIMEOUT = 30000;
+  private static final long DEFAULT_CONNECTION_REQUEST_TIMEOUT = 30 * 1000;
 
   // The default time to wait for data, the maximum time between two consecutive packets of data.
-  private static final int DEFAULT_RESPONSE_TIMEOUT = 60000;
+  private static final int DEFAULT_RESPONSE_TIMEOUT = 60 * 1000;
 
   // The default time to keep a connection alive.
   private static final long DEFAULT_KEEP_ALIVE_TIME_MILLIS = 20 * 1000;
 
   private ZitiConnectionSocketFactory zitiConnectionSocketFactory;
   private ZitiSSLConnectionSocketFactory zitiSSLConnectionSocketFactory;
 
-  @ConditionalOnMissingBean(name = "zitiRestTemplate")
-  @Bean("zitiRestTemplate")
-  public RestTemplate restTemplate(@Qualifier("zitiRestTemplateBuilder") RestTemplateBuilder restTemplateBuilder) {
+  @ConditionalOnProperty(value = "spring.ziti.client.rest-template.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public RestTemplate zitiRestTemplate(@Qualifier("zitiRestTemplateBuilder") RestTemplateBuilder restTemplateBuilder) {
```

Comment:
adding ziti everywhere to me seems unnecessary. it's the ziti jvm sdk. it's necessary in properties maybe but not in the internal code?

Reply:
It's intentional, the Spring context is global, without a naming prefix it clutters up the entire context and would make it conflict with other beans. For example the "restTemplateBuilder" is a bean already defined, this defines the "zitiRestTemplateBuilder" so that applications can use both.